app/com/github/pedrovgs/dotto/core/algebra/dotto.scala

```diff
@@ -0,0 +1,27 @@
+package com.github.pedrovgs.dotto.core.algebra
+
+import cats.free.Free
+import com.github.pedrovgs.dotto.core.types.{LedInteraction, MorseSymbol}
+
+/**
+  * Application algebra used to describe what our program does. Implemented using Cats as core library.
+  */
+object dotto {
+
+  type Message = String
+  type MorseSentence = Seq[Seq[MorseSymbol]]
+  type LedInteractions = Seq[LedInteraction]
+
+  sealed trait Instruction[A]
```

Comment:
People often model ADTs with sealed traits but a more efficient bytecode of doing this is using `abstract class ... extends Serializable with Product` This will also aid the compiler with inference.